### ISSUE TITLE
Created a C frontend for QFTASM - License compatibility?

### ISSUE THREAD

**Issue by woodrush** [QuestForTetris/WriteUp#8]
Hi everyone, I wanted to ask a question about the license compatibility of a QFTASM-related project.

I have just finished my project on using [ELVM](https://github.com/shinh/elvm) to create a toolchain path that can compile C code to QFTASM. ELVM (short for Esoteric-Language Virtual Machine) is a virtual processor architecture inspired by LLVM, that was created to port C code to Esoteric Programming Languages such as Unlambda and Piet. On my forked repository (https://github.com/woodrush/elvm/tree/add-qftasm) is a QFTASM backend for ELVM, which completes the path for compiling C code to QFTASM. I also wrote a QFTASM interpreter in Python for running tests on the ELVM repository.

I also carefully studied codes in the [QFT repository](https://github.com/QuestForTetris/QFT), and managed to extend the ROM and RAM in [Tetris.mc](https://github.com/QuestForTetris/QFT/blob/master/Tetris.mc) to successfully run a [program written in C](https://github.com/woodrush/elvm/blob/add-qftasm/tools/qftasm/calc.c) on the ROM-RAM-extended version of Tetris.mc. While the code is available, I haven't put the Varlife pattern in the repository yet.

Now, I noticed that while the main repository containing Tetris.mc does not have a license associated with it, the repos for the [qftasm-interpreter](https://github.com/QuestForTetris/qftasm-interpreter) and the [Varlife simulator](https://github.com/QuestForTetris/variations-of-life) each has an MIT License.

My question is, would it therefore be possible to publish a pull request for my current QFTASM-related project to the original ELVM repository so that it could be merged? The ELVM repository has an MIT License as well.

Also, would it be OK if I published code related to the actual Tetris.mc Varlife pattern, such as ROM creation scripts? Would it be possible for a license to be associated with the repository containing the pattern so that it could be made Open Source? It was a concern for me since the repository containing it didn't have a license.

Finally, I want to say that I was astonished by the odyssey of The Quest for Tetris project. I was especially amazed by how closely the final architecture resembled that of physical, electrical circuits, while at the same time being constructed upon the rules and atoms of The Game of Life. I respect the effort that has been made to create such an inspiring work.

(Edit: grammar)

**Comment by woodrush** [comment]
I'd like to ping @Mego @muddyfish who seems to be active recently on GitHub and seems not to be watching (receiving notifications on) this repo.

**Comment by muddyfish** [comment]
Hi there. That sounds really cool, but I've not been involved in any of the code related to that repository so I can't really be of help with licensing issues, though I can poke this issue elsewhere

**Comment by woodrush** [comment]
Hi @muddyfish, thanks for your quick reply. I'd be really happy if you could do that, could you bring it up to the members related to that repository?

**Comment by elendiastarman** [comment]
:wave: I am not a lawyer etc etc but I picked the MIT license for the QFTASM interpreter and VarLife repos specifically to encourage sharing and building on the work here, and I suspect everyone on this project would feel similarly about making it as available as possible. That said, we do need a license, so I'm gonna create a pull request.

**Comment by elendiastarman** [comment]
https://github.com/QuestForTetris/QFT/pull/12

**Comment by woodrush** [comment]
@elendiastarman Thanks for the super-fast PR! I'll be working on the pull request on my side too once it gets merged.

**Comment by woodrush** [comment]
Hi, I have some updates on this issue. First, we got the QFTASM project [merged](https://github.com/shinh/elvm/pull/103) into ELVM, so we're able to run C on QFTASM now! Thanks again to elandiastarman for your support on the License. I also made the [QFT-devkit](https://github.com/woodrush/QFT-devkit) to put the tools I used for experimenting with QFTASM.

Also excitingly, as a derivative project, I made [lisp-in-life](https://github.com/woodrush/lisp-in-life), where I had a Lisp interpreter successfully work in Varlife using the QFT computer. The ROM is 9632 bytes long, and uses 512 bytes of RAM. I'm now on my way on running the metafied version of the Varlife pattern (Golly has been running for 33 hours now, haha).

I'm closing this issue since I believe we're all set with the Licenses. Thanks again for your support, and such a cool project!
